src/Renderer/Hal.php

```diff
@@ -17,55 +23,57 @@
  * @see         http://stateless.co/hal_specification.html
  * @since       1.2
  */
-class JApiHalDocumentDocument extends JDocument
+class Hal extends Base
 {
 	/**
-	 * Document name
-	 *
-	 * @var    string
-	 * @since  1.2
-	 */
-	protected $name = 'joomla';
-
-	/**
 	 * Render all hrefs as absolute, relative is default
+	 *
+	 * @var    boolean
+	 * @since  __DELPOY_VERSION__
 	 */
 	protected $absoluteHrefs = false;
 
 	/**
 	 * Document format (xml or json)
+	 *
+	 * @var    string
+	 * @since  __DEPLOY__VERSION__
 	 */
-	protected $documentFormat = false;
+	protected $documentFormat;
 
 	/**
-	 * @var    JApiHalHal  Main HAL object
+	 * Webservice object
+	 *
+	 * @var    Webservice
 	 * @since  1.2
 	 */
-	public $hal = null;
+	public $webservice = null;
 
 	/**
 	 * Class constructor
 	 *
-	 * @param   array  $options  Associative array of options
+	 * @param   Container  $container  The DIC object
+	 * @param   array      $options    Associative array of options
 	 *
 	 * @since  1.2
 	 */
-	public function __construct($options = array())
+	public function __construct(Container $container, $options = array())
 	{
-		parent::__construct($options);
+		parent::__construct($container, $options);
 
 		$this->documentFormat = $options['documentFormat'];
 
+		// Sanity check - fall back to XML format
 		if (!in_array($this->documentFormat, array('xml', 'json')))
 		{
 			$this->documentFormat = 'json';
 		}
 
 		// Set default mime type.
-		$this->_mime = 'application/hal+' . $this->documentFormat;
+		$this->setMimeEncoding('application/hal+' . $this->documentFormat, false);
 
 		// Set document type.
-		$this->_type = 'hal+' . $this->documentFormat;
+		$this->setType('hal+' . $this->documentFormat);
 
 		// Set absolute/relative hrefs.
 		$this->absoluteHrefs = isset($options['absoluteHrefs']) ? $options['absoluteHrefs'] : true;
```

Comment:
I'd suggest you merge that into the current version and then I can merge staging into this